pkg/ingester/retention.go

```diff
@@ -371,6 +375,11 @@ func (bm *realFSBlockManager) GetTenantIDs(ctx context.Context) ([]string, error
 		if !tenantDir.IsDir() {
 			continue
 		}
+
+		if tenantDir.Name() == lostAndFoundDirName {
```

Comment:
i'm slightly worried that `lost+found` is not the last directory that might cause issues here. Instead of having a blocklist of directories, that we have to keep up to date, can we go the opposite way and find some heuristic that would tell us that a directory is a "pyroscope data directory"?

Reply:
I guess not having a `local` subfolder might be a good enough heuristic to skip

Reply:
Since we are ultimately going to look into the local folder to delete blocks that would be a good start. Can we also assume tenant ids are going to be integers? That would allow us to also skip directories which aren't integer names.

Reply:
we do have the `anonymous` tenant as well, though we could skip (or add manually)

Reply:
I think tenant ids can be anything.

> I guess not having a local subfolder might be a good enough heuristic to skip

this sounds like the best one

Reply:
This is fixed in https://github.com/grafana/pyroscope/pull/2916/commits/a952c4d17e14c771acad7afeaad50598fc924484. The `isTenantDir` function holds the logic to determine if a directory is a tenant or not.